pkgs/development/python-modules/gdsfactory/default.nix

```
@@ -0,0 +1,105 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pytestCheckHook,
+  flit-core,
+  jinja2,
+  loguru,
+  matplotlib,
+  numpy,
+  orjson,
+  pandas,
+  pydantic,
+  pydantic-settings,
+  pydantic-extra-types,
+  pyyaml,
+  qrcode,
+  rectpack,
+  rich,
+  scipy,
+  shapely,
+  toolz,
+  types-pyyaml,
+  typer,
+  kfactory,
+  watchdog,
+  freetype-py,
+  mapbox-earcut,
+  networkx,
+  scikit-image,
+  trimesh,
+  ipykernel,
+  attrs,
+  graphviz,
+  # tests
+  jsondiff,
+  jsonschema,
+  pytest-regressions,
+}:
+buildPythonPackage rec {
+  pname = "gdsfactory";
+  version = "8.18.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "gdsfactory";
+    repo = "gdsfactory";
+    rev = "v${version}";
+    hash = "sha256-wDz8QpRgu40FB8+otnGsHVn2e6/SWXIZgA1aeMqMhPQ=";
+  };
+
+  build-system = [ flit-core ];
+
+  dependencies = [
+    jinja2
+    loguru
+    matplotlib
+    numpy
+    orjson
+    pandas
+    pydantic
+    pydantic-settings
+    pydantic-extra-types
+    pyyaml
+    qrcode
+    rectpack
+    rich
+    scipy
+    shapely
+    toolz
+    types-pyyaml
+    typer
+    kfactory
+    watchdog
+    freetype-py
+    mapbox-earcut
+    networkx
+    scikit-image
+    trimesh
+    ipykernel
+    attrs
+    graphviz
+  ];
+
+  nativeCheckInputs = [
+    jsondiff
+    jsonschema
+    pytestCheckHook
+    pytest-regressions
+  ];
+
+  pythonRelaxDeps = [ "pydantic" ];
```

```suggestion
  pythonRelaxDeps = [
    "pydantic"
    "trimesh"
  ];
```
nixpkgs has trimesh 4.6.2, package expects <4.6
(also check if it still works with the newer version)

---

With the proposed change, it seems to work fine.